pkg/kubernetes/details.js

```diff
@@ -3,12 +3,130 @@ define([
     "base1/cockpit",
     "base1/angular",
     "base1/term",
+    "kubernetes/client",
     "kubernetes/app"
-], function($, cockpit, angular, Terminal) {
+], function($, cockpit, angular, Terminal, kubernetes) {
     'use strict';
 
     var _ = cockpit.gettext;
 
+    var k8client =  kubernetes.k8client();
+    var adjust_btn = $('#adjust-entity-button'); 
+    var adjust_entity_dlg = $('#adjust-entity-dialog');
+    var delete_btn = $('#delete-entity-button'); 
+    var delete_entity_dlg = $('#delete-entity-dialog');
+    var delete_pod_dlg = $('#delete-pod-dialog');
+    var delete_pod_btn = $('#delete-pod-button');
+
+    delete_entity_dlg.on('show.bs.modal', function(e) {
+        var name = $(e.relatedTarget).attr("data-name");
+        var kind = $(e.relatedTarget).attr("data-kind");
+        var ns = $(e.relatedTarget).attr("data-ns");
+        var entity = k8client.lookup(kind, name, ns);
+        delete_entity_dlg.find('.modal-body').text("Delete " + kind + " " + name + "?");
+
+        delete_btn.on('click', function() {
+            k8client.remove(entity.metadata.selfLink);
+            delete_entity_dlg.modal('hide');
+        });
+    });
+
+    adjust_entity_dlg.on('show.bs.modal', function(e) {
+        var name = $(e.relatedTarget).attr("data-name");
+        var kind = $(e.relatedTarget).attr("data-kind");
+        var ns = $(e.relatedTarget).attr("data-ns");
+        var entity = k8client.lookup(kind, name, ns);
+
+        adjust_entity_dlg.find('#replica-count').val(entity.spec.replicas);
+
+        adjust_btn.on('click', function() {
+            function resize(item, value) {
+                var spec = item.spec;
+                if (!spec) {
+                    console.warn("replicationcontroller without spec");
+                    return false;
+                }
+
+                /* Already set at same value */
+                if (spec.replicas === value)
+                    return false;
+
+                spec.replicas = value;
+                return true;
+            }
+
+            function update_replica(input, rc) {
+                var failures = [];
+                var dfd = $.Deferred();
+                var req;
+                var ex;
+
+                var value = Number($.trim(input));
+                if (isNaN(value) || value < 0)
+                    ex = new Error(_("Not a valid number of replicas"));
+                else if (value > 128)
+                    ex = new Error(_("The maximum number of replicas is 128"));
+
+                if (ex) {
+                    ex.target = "#replica-count";
+                    failures.push(ex);
+                }
+
+                var task = function() {
```

Comment:
Why are we defining a task for a single action? It seems there's no need for this abstraction here.